src/main/java/uk/co/onsdigital/discovery/metadata/api/service/MetadataServiceImpl.java

```diff
@@ -59,14 +62,20 @@ public DataSet findDataSetById(String dataSetId) throws DataSetNotFoundException
 
     @Transactional(readOnly = true)
     public Set<Dimension> listDimensionsForDataSet(String dataSetId) throws DataSetNotFoundException {
-        final Set<ConceptSystem> concepts = metadataDao.findConceptSystemsInDataSet(dataSetId);
-        return convertConceptSystemsToDimensions(concepts, dataSetId, true);
+        final DimensionalDataSet dataSet = metadataDao.findDataSetById(dataSetId);
+        return convertAllDimensions(dataSet, true);
     }
 
     @Transactional(readOnly = true)
     public Dimension findDimensionById(String dataSetId, String dimensionId) throws DataSetNotFoundException, DimensionNotFoundException {
-        final ConceptSystem conceptSystem = metadataDao.findConceptSystemByDataSetAndConceptSystemName(dataSetId, dimensionId);
-        return convertConceptSystemToDimension(conceptSystem, dataSetId, true);
+        try {
+            final ConceptSystem conceptSystem = metadataDao.findConceptSystemByDataSetAndConceptSystemName(dataSetId, dimensionId);
+            return convertConceptSystemToDimension(conceptSystem, dataSetId, true);
```

Comment:
We seem to use the terms dimension and concept system interchangeably which could be confusing. Do you think its worth standardising which we use throughout?